### ISSUE TITLE
error: unrecognized command line option ‘-std=c++14’

### ISSUE THREAD

**Issue by jihunni** [Boyle-Lab/Blacklist#27]
I followed your instruction in README and tried to build Blacklist. When I tried to build Blacklist, an error message pops up.
```
# make
g++ -std=c++14 -o Blacklist blacklist.cpp -I/opt/The_ENCODE_Blacklist/20210922/bamtools/install/include/bamtools -L/opt/The_ENCODE_Blacklist/20210922/bamtools/install/lib/bamtools -lbamtools -lz -Wl,-rpath,/opt/The_ENCODE_Blacklist/20210922/bamtools/install/lib/bamtools
g++: error: unrecognized command line option ‘-std=c++14’
make: *** [all] Error 1
```

Could you give me any instructions to solve this error? If you need more information about my environment, please leave what information you need in a comment. Then I will reply to it.

Thank you in advance!

**Comment by jihunni** [comment]
In addition, I found an article in stack overflow  (https://stackoverflow.com/questions/36245428/c-error-unrecognized-command-line-option-std-c14). I changed in Makefile from
```
all:
        g++ -std=c++14 -o Blacklist blacklist.cpp -I$(BAMTOOLS_INCLUDE_DIR) -L$(BAMTOOLS_LIB_DIR) -lbamtools -lz -Wl,-rpath,$(BAMTOOLS_LIB_DIR)
```
into
```
all:
        g++ -std=c++1y -o Blacklist blacklist.cpp -I$(BAMTOOLS_INCLUDE_DIR) -L$(BAMTOOLS_LIB_DIR) -lbamtools -lz -Wl,-rpath,$(BAMTOOLS_LIB_DIR)
```

But I have another error message

```
[root@life1 20210922]# make
g++ -std=c++1y -o Blacklist blacklist.cpp -I/opt/The_ENCODE_Blacklist/20210922/bamtools/install/include/bamtools -L/opt/The_ENCODE_Blacklist/20210922/bamtools/install/lib/bamtools -lbamtools -lz -Wl,-rpath,/opt/The_ENCODE_Blacklist/20210922/bamtools/install/lib/bamtools
blacklist.cpp: In function ‘void rankify(std::vector<double>&)’:
blacklist.cpp:216:45: error: parameter declared ‘auto’
  std::sort(begin(T), end(T), [](auto const &t1, auto const &t2) {
                                             ^
blacklist.cpp:216:61: error: parameter declared ‘auto’
  std::sort(begin(T), end(T), [](auto const &t1, auto const &t2) {
                                                             ^
blacklist.cpp: In lambda function:
blacklist.cpp:217:24: error: ‘t1’ was not declared in this scope
          return get<0>(t1) < get<0>(t2); // or use a custom compare function
                        ^
blacklist.cpp:217:37: error: ‘t2’ was not declared in this scope
          return get<0>(t1) < get<0>(t2); // or use a custom compare function
                                     ^
blacklist.cpp: In function ‘std::vector<double> getAbnormalRegions(std::vector<SequenceData>, std::vector<int>, int, bool)’:
blacklist.cpp:335:22: error: call of overloaded ‘isnan(double&)’ is ambiguous
    if(!isnan(normTemp) && !isinf(normTemp) && normTemp > 0) {
                      ^
blacklist.cpp:335:22: note: candidates are:
In file included from /usr/include/features.h:375:0,
                 from /usr/include/c++/4.8.2/x86_64-redhat-linux/bits/os_defines.h:39,
                 from /usr/include/c++/4.8.2/x86_64-redhat-linux/bits/c++config.h:2097,
                 from /usr/include/c++/4.8.2/iostream:38,
                 from blacklist.cpp:11:
/usr/include/bits/mathcalls.h:235:1: note: int isnan(double)
 __MATHDECL_1 (int,isnan,, (_Mdouble_ __value)) __attribute__ ((__const__));
 ^
In file included from blacklist.cpp:15:0:
/usr/include/c++/4.8.2/cmath:626:3: note: constexpr bool std::isnan(long double)
   isnan(long double __x)
   ^
/usr/include/c++/4.8.2/cmath:622:3: note: constexpr bool std::isnan(double)
   isnan(double __x)
   ^
/usr/include/c++/4.8.2/cmath:618:3: note: constexpr bool std::isnan(float)
   isnan(float __x)
   ^
blacklist.cpp:335:42: error: call of overloaded ‘isinf(double&)’ is ambiguous
    if(!isnan(normTemp) && !isinf(normTemp) && normTemp > 0) {
                                          ^
blacklist.cpp:335:42: note: candidates are:
In file included from /usr/include/features.h:375:0,
                 from /usr/include/c++/4.8.2/x86_64-redhat-linux/bits/os_defines.h:39,
                 from /usr/include/c++/4.8.2/x86_64-redhat-linux/bits/c++config.h:2097,
                 from /usr/include/c++/4.8.2/iostream:38,
                 from blacklist.cpp:11:
/usr/include/bits/mathcalls.h:202:1: note: int isinf(double)
 __MATHDECL_1 (int,isinf,, (_Mdouble_ __value)) __attribute__ ((__const__));
 ^
In file included from blacklist.cpp:15:0:
/usr/include/c++/4.8.2/cmath:608:3: note: constexpr bool std::isinf(long double)
   isinf(long double __x)
   ^
/usr/include/c++/4.8.2/cmath:604:3: note: constexpr bool std::isinf(double)
   isinf(double __x)
   ^
/usr/include/c++/4.8.2/cmath:600:3: note: constexpr bool std::isinf(float)
   isinf(float __x)
   ^
In file included from /usr/include/c++/4.8.2/algorithm:62:0,
                 from blacklist.cpp:18:
/usr/include/c++/4.8.2/bits/stl_algo.h: In instantiation of ‘void std::__insertion_sort(_RandomAccessIterator, _RandomAccessIterator, _Compare) [with _RandomAccessIterator = __gnu_cxx::__normal_iterator<std::tuple<double, int>*, std::vector<std::tuple<double, int> > >; _Compare = rankify(std::vector<double>&)::__lambda0]’:
/usr/include/c++/4.8.2/bits/stl_algo.h:2226:70:   required from ‘void std::__final_insertion_sort(_RandomAccessIterator, _RandomAccessIterator, _Compare) [with _RandomAccessIterator = __gnu_cxx::__normal_iterator<std::tuple<double, int>*, std::vector<std::tuple<double, int> > >; _Compare = rankify(std::vector<double>&)::__lambda0]’
/usr/include/c++/4.8.2/bits/stl_algo.h:5500:55:   required from ‘void std::sort(_RAIter, _RAIter, _Compare) [with _RAIter = __gnu_cxx::__normal_iterator<std::tuple<double, int>*, std::vector<std::tuple<double, int> > >; _Compare = rankify(std::vector<double>&)::__lambda0]’
blacklist.cpp:218:3:   required from here
/usr/include/c++/4.8.2/bits/stl_algo.h:2159:29: error: no match for call to ‘(rankify(std::vector<double>&)::__lambda0) (std::tuple<double, int>&, std::tuple<double, int>&)’
    if (__comp(*__i, *__first))
                             ^
blacklist.cpp:216:31: note: candidates are:
  std::sort(begin(T), end(T), [](auto const &t1, auto const &t2) {
                               ^
In file included from /usr/include/c++/4.8.2/algorithm:62:0,
                 from blacklist.cpp:18:
/usr/include/c++/4.8.2/bits/stl_algo.h:2159:29: note: void (*)() <conversion>
    if (__comp(*__i, *__first))
                             ^
/usr/include/c++/4.8.2/bits/stl_algo.h:2159:29: note:   candidate expects 1 argument, 3 provided
blacklist.cpp:216:63: note: rankify(std::vector<double>&)::__lambda0
  std::sort(begin(T), end(T), [](auto const &t1, auto const &t2) {
                                                               ^
blacklist.cpp:216:63: note:   candidate expects 0 arguments, 2 provided
In file included from /usr/include/c++/4.8.2/algorithm:62:0,
                 from blacklist.cpp:18:
/usr/include/c++/4.8.2/bits/stl_algo.h: In instantiation of ‘void std::__heap_select(_RandomAccessIterator, _RandomAccessIterator, _RandomAccessIterator, _Compare) [with _RandomAccessIterator = __gnu_cxx::__normal_iterator<std::tuple<double, int>*, std::vector<std::tuple<double, int> > >; _Compare = rankify(std::vector<double>&)::__lambda0]’:
/usr/include/c++/4.8.2/bits/stl_algo.h:5349:59:   required from ‘void std::partial_sort(_RAIter, _RAIter, _RAIter, _Compare) [with _RAIter = __gnu_cxx::__normal_iterator<std::tuple<double, int>*, std::vector<std::tuple<double, int> > >; _Compare = rankify(std::vector<double>&)::__lambda0]’
/usr/include/c++/4.8.2/bits/stl_algo.h:2332:68:   required from ‘void std::__introsort_loop(_RandomAccessIterator, _RandomAccessIterator, _Size, _Compare) [with _RandomAccessIterator = __gnu_cxx::__normal_iterator<std::tuple<double, int>*, std::vector<std::tuple<double, int> > >; _Size = long int; _Compare = rankify(std::vector<double>&)::__lambda0]’
/usr/include/c++/4.8.2/bits/stl_algo.h:5499:44:   required from ‘void std::sort(_RAIter, _RAIter, _Compare) [with _RAIter = __gnu_cxx::__normal_iterator<std::tuple<double, int>*, std::vector<std::tuple<double, int> > >; _Compare = rankify(std::vector<double>&)::__lambda0]’
blacklist.cpp:218:3:   required from here
/usr/include/c++/4.8.2/bits/stl_algo.h:1948:27: error: no match for call to ‘(rankify(std::vector<double>&)::__lambda0) (std::tuple<double, int>&, std::tuple<double, int>&)’
  if (__comp(*__i, *__first))
                           ^
blacklist.cpp:216:31: note: candidates are:
  std::sort(begin(T), end(T), [](auto const &t1, auto const &t2) {
                               ^
In file included from /usr/include/c++/4.8.2/algorithm:62:0,
                 from blacklist.cpp:18:
/usr/include/c++/4.8.2/bits/stl_algo.h:1948:27: note: void (*)() <conversion>
  if (__comp(*__i, *__first))
                           ^
/usr/include/c++/4.8.2/bits/stl_algo.h:1948:27: note:   candidate expects 1 argument, 3 provided
blacklist.cpp:216:63: note: rankify(std::vector<double>&)::__lambda0
  std::sort(begin(T), end(T), [](auto const &t1, auto const &t2) {
                                                               ^
blacklist.cpp:216:63: note:   candidate expects 0 arguments, 2 provided
In file included from /usr/include/c++/4.8.2/algorithm:62:0,
                 from blacklist.cpp:18:
/usr/include/c++/4.8.2/bits/stl_algo.h: In instantiation of ‘void std::__move_median_to_first(_Iterator, _Iterator, _Iterator, _Iterator, _Compare) [with _Iterator = __gnu_cxx::__normal_iterator<std::tuple<double, int>*, std::vector<std::tuple<double, int> > >; _Compare = rankify(std::vector<double>&)::__lambda0]’:
/usr/include/c++/4.8.2/bits/stl_algo.h:2295:13:   required from ‘_RandomAccessIterator std::__unguarded_partition_pivot(_RandomAccessIterator, _RandomAccessIterator, _Compare) [with _RandomAccessIterator = __gnu_cxx::__normal_iterator<std::tuple<double, int>*, std::vector<std::tuple<double, int> > >; _Compare = rankify(std::vector<double>&)::__lambda0]’
/usr/include/c++/4.8.2/bits/stl_algo.h:2337:62:   required from ‘void std::__introsort_loop(_RandomAccessIterator, _RandomAccessIterator, _Size, _Compare) [with _RandomAccessIterator = __gnu_cxx::__normal_iterator<std::tuple<double, int>*, std::vector<std::tuple<double, int> > >; _Size = long int; _Compare = rankify(std::vector<double>&)::__lambda0]’
/usr/include/c++/4.8.2/bits/stl_algo.h:5499:44:   required from ‘void std::sort(_RAIter, _RAIter, _Compare) [with _RAIter = __gnu_cxx::__normal_iterator<std::tuple<double, int>*, std::vector<std::tuple<double, int> > >; _Compare = rankify(std::vector<double>&)::__lambda0]’
blacklist.cpp:218:3:   required from here
/usr/include/c++/4.8.2/bits/stl_algo.h:114:28: error: no match for call to ‘(rankify(std::vector<double>&)::__lambda0) (std::tuple<double, int>&, std::tuple<double, int>&)’
       if (__comp(*__a, *__b))
                            ^
blacklist.cpp:216:31: note: candidates are:
  std::sort(begin(T), end(T), [](auto const &t1, auto const &t2) {
                               ^
In file included from /usr/include/c++/4.8.2/algorithm:62:0,
                 from blacklist.cpp:18:
/usr/include/c++/4.8.2/bits/stl_algo.h:114:28: note: void (*)() <conversion>
       if (__comp(*__a, *__b))
                            ^
/usr/include/c++/4.8.2/bits/stl_algo.h:114:28: note:   candidate expects 1 argument, 3 provided
blacklist.cpp:216:63: note: rankify(std::vector<double>&)::__lambda0
  std::sort(begin(T), end(T), [](auto const &t1, auto const &t2) {
                                                               ^
blacklist.cpp:216:63: note:   candidate expects 0 arguments, 2 provided
In file included from /usr/include/c++/4.8.2/algorithm:62:0,
                 from blacklist.cpp:18:
/usr/include/c++/4.8.2/bits/stl_algo.h:116:25: error: no match for call to ‘(rankify(std::vector<double>&)::__lambda0) (std::tuple<double, int>&, std::tuple<double, int>&)’
    if (__comp(*__b, *__c))
                         ^
blacklist.cpp:216:31: note: candidates are:
  std::sort(begin(T), end(T), [](auto const &t1, auto const &t2) {
                               ^
In file included from /usr/include/c++/4.8.2/algorithm:62:0,
                 from blacklist.cpp:18:
/usr/include/c++/4.8.2/bits/stl_algo.h:116:25: note: void (*)() <conversion>
    if (__comp(*__b, *__c))
                         ^
/usr/include/c++/4.8.2/bits/stl_algo.h:116:25: note:   candidate expects 1 argument, 3 provided
blacklist.cpp:216:63: note: rankify(std::vector<double>&)::__lambda0
  std::sort(begin(T), end(T), [](auto const &t1, auto const &t2) {
                                                               ^
blacklist.cpp:216:63: note:   candidate expects 0 arguments, 2 provided
In file included from /usr/include/c++/4.8.2/algorithm:62:0,
                 from blacklist.cpp:18:
/usr/include/c++/4.8.2/bits/stl_algo.h:118:30: error: no match for call to ‘(rankify(std::vector<double>&)::__lambda0) (std::tuple<double, int>&, std::tuple<double, int>&)’
    else if (__comp(*__a, *__c))
                              ^
blacklist.cpp:216:31: note: candidates are:
  std::sort(begin(T), end(T), [](auto const &t1, auto const &t2) {
                               ^
In file included from /usr/include/c++/4.8.2/algorithm:62:0,
                 from blacklist.cpp:18:
/usr/include/c++/4.8.2/bits/stl_algo.h:118:30: note: void (*)() <conversion>
    else if (__comp(*__a, *__c))
                              ^
/usr/include/c++/4.8.2/bits/stl_algo.h:118:30: note:   candidate expects 1 argument, 3 provided
blacklist.cpp:216:63: note: rankify(std::vector<double>&)::__lambda0
  std::sort(begin(T), end(T), [](auto const &t1, auto const &t2) {
                                                               ^
blacklist.cpp:216:63: note:   candidate expects 0 arguments, 2 provided
In file included from /usr/include/c++/4.8.2/algorithm:62:0,
                 from blacklist.cpp:18:
/usr/include/c++/4.8.2/bits/stl_algo.h:123:33: error: no match for call to ‘(rankify(std::vector<double>&)::__lambda0) (std::tuple<double, int>&, std::tuple<double, int>&)’
       else if (__comp(*__a, *__c))
                                 ^
blacklist.cpp:216:31: note: candidates are:
  std::sort(begin(T), end(T), [](auto const &t1, auto const &t2) {
                               ^
In file included from /usr/include/c++/4.8.2/algorithm:62:0,
                 from blacklist.cpp:18:
/usr/include/c++/4.8.2/bits/stl_algo.h:123:33: note: void (*)() <conversion>
       else if (__comp(*__a, *__c))
                                 ^
/usr/include/c++/4.8.2/bits/stl_algo.h:123:33: note:   candidate expects 1 argument, 3 provided
blacklist.cpp:216:63: note: rankify(std::vector<double>&)::__lambda0
  std::sort(begin(T), end(T), [](auto const &t1, auto const &t2) {
                                                               ^
blacklist.cpp:216:63: note:   candidate expects 0 arguments, 2 provided
In file included from /usr/include/c++/4.8.2/algorithm:62:0,
                 from blacklist.cpp:18:
/usr/include/c++/4.8.2/bits/stl_algo.h:125:33: error: no match for call to ‘(rankify(std::vector<double>&)::__lambda0) (std::tuple<double, int>&, std::tuple<double, int>&)’
       else if (__comp(*__b, *__c))
                                 ^
blacklist.cpp:216:31: note: candidates are:
  std::sort(begin(T), end(T), [](auto const &t1, auto const &t2) {
                               ^
In file included from /usr/include/c++/4.8.2/algorithm:62:0,
                 from blacklist.cpp:18:
/usr/include/c++/4.8.2/bits/stl_algo.h:125:33: note: void (*)() <conversion>
       else if (__comp(*__b, *__c))
                                 ^
/usr/include/c++/4.8.2/bits/stl_algo.h:125:33: note:   candidate expects 1 argument, 3 provided
blacklist.cpp:216:63: note: rankify(std::vector<double>&)::__lambda0
  std::sort(begin(T), end(T), [](auto const &t1, auto const &t2) {
                                                               ^
blacklist.cpp:216:63: note:   candidate expects 0 arguments, 2 provided
In file included from /usr/include/c++/4.8.2/algorithm:62:0,
                 from blacklist.cpp:18:
/usr/include/c++/4.8.2/bits/stl_algo.h: In instantiation of ‘_RandomAccessIterator std::__unguarded_partition(_RandomAccessIterator, _RandomAccessIterator, const _Tp&, _Compare) [with _RandomAccessIterator = __gnu_cxx::__normal_iterator<std::tuple<double, int>*, std::vector<std::tuple<double, int> > >; _Tp = std::tuple<double, int>; _Compare = rankify(std::vector<double>&)::__lambda0]’:
/usr/include/c++/4.8.2/bits/stl_algo.h:2296:78:   required from ‘_RandomAccessIterator std::__unguarded_partition_pivot(_RandomAccessIterator, _RandomAccessIterator, _Compare) [with _RandomAccessIterator = __gnu_cxx::__normal_iterator<std::tuple<double, int>*, std::vector<std::tuple<double, int> > >; _Compare = rankify(std::vector<double>&)::__lambda0]’
/usr/include/c++/4.8.2/bits/stl_algo.h:2337:62:   required from ‘void std::__introsort_loop(_RandomAccessIterator, _RandomAccessIterator, _Size, _Compare) [with _RandomAccessIterator = __gnu_cxx::__normal_iterator<std::tuple<double, int>*, std::vector<std::tuple<double, int> > >; _Size = long int; _Compare = rankify(std::vector<double>&)::__lambda0]’
/usr/include/c++/4.8.2/bits/stl_algo.h:5499:44:   required from ‘void std::sort(_RAIter, _RAIter, _Compare) [with _RAIter = __gnu_cxx::__normal_iterator<std::tuple<double, int>*, std::vector<std::tuple<double, int> > >; _Compare = rankify(std::vector<double>&)::__lambda0]’
blacklist.cpp:218:3:   required from here
/usr/include/c++/4.8.2/bits/stl_algo.h:2263:35: error: no match for call to ‘(rankify(std::vector<double>&)::__lambda0) (std::tuple<double, int>&, const std::tuple<double, int>&)’
    while (__comp(*__first, __pivot))
                                   ^
blacklist.cpp:216:31: note: candidates are:
  std::sort(begin(T), end(T), [](auto const &t1, auto const &t2) {
                               ^
In file included from /usr/include/c++/4.8.2/algorithm:62:0,
                 from blacklist.cpp:18:
/usr/include/c++/4.8.2/bits/stl_algo.h:2263:35: note: void (*)() <conversion>
    while (__comp(*__first, __pivot))
                                   ^
/usr/include/c++/4.8.2/bits/stl_algo.h:2263:35: note:   candidate expects 1 argument, 3 provided
blacklist.cpp:216:63: note: rankify(std::vector<double>&)::__lambda0
  std::sort(begin(T), end(T), [](auto const &t1, auto const &t2) {
                                                               ^
blacklist.cpp:216:63: note:   candidate expects 0 arguments, 2 provided
In file included from /usr/include/c++/4.8.2/algorithm:62:0,
                 from blacklist.cpp:18:
/usr/include/c++/4.8.2/bits/stl_algo.h:2266:34: error: no match for call to ‘(rankify(std::vector<double>&)::__lambda0) (const std::tuple<double, int>&, std::tuple<double, int>&)’
    while (__comp(__pivot, *__last))
                                  ^
blacklist.cpp:216:31: note: candidates are:
  std::sort(begin(T), end(T), [](auto const &t1, auto const &t2) {
                               ^
In file included from /usr/include/c++/4.8.2/algorithm:62:0,
                 from blacklist.cpp:18:
/usr/include/c++/4.8.2/bits/stl_algo.h:2266:34: note: void (*)() <conversion>
    while (__comp(__pivot, *__last))
                                  ^
/usr/include/c++/4.8.2/bits/stl_algo.h:2266:34: note:   candidate expects 1 argument, 3 provided
blacklist.cpp:216:63: note: rankify(std::vector<double>&)::__lambda0
  std::sort(begin(T), end(T), [](auto const &t1, auto const &t2) {
                                                               ^
blacklist.cpp:216:63: note:   candidate expects 0 arguments, 2 provided
In file included from /usr/include/c++/4.8.2/bits/stl_algo.h:61:0,
                 from /usr/include/c++/4.8.2/algorithm:62,
                 from blacklist.cpp:18:
/usr/include/c++/4.8.2/bits/stl_heap.h: In instantiation of ‘void std::__adjust_heap(_RandomAccessIterator, _Distance, _Distance, _Tp, _Compare) [with _RandomAccessIterator = __gnu_cxx::__normal_iterator<std::tuple<double, int>*, std::vector<std::tuple<double, int> > >; _Distance = long int; _Tp = std::tuple<double, int>; _Compare = rankify(std::vector<double>&)::__lambda0]’:
/usr/include/c++/4.8.2/bits/stl_heap.h:448:15:   required from ‘void std::make_heap(_RAIter, _RAIter, _Compare) [with _RAIter = __gnu_cxx::__normal_iterator<std::tuple<double, int>*, std::vector<std::tuple<double, int> > >; _Compare = rankify(std::vector<double>&)::__lambda0]’
/usr/include/c++/4.8.2/bits/stl_algo.h:1946:47:   required from ‘void std::__heap_select(_RandomAccessIterator, _RandomAccessIterator, _RandomAccessIterator, _Compare) [with _RandomAccessIterator = __gnu_cxx::__normal_iterator<std::tuple<double, int>*, std::vector<std::tuple<double, int> > >; _Compare = rankify(std::vector<double>&)::__lambda0]’
/usr/include/c++/4.8.2/bits/stl_algo.h:5349:59:   required from ‘void std::partial_sort(_RAIter, _RAIter, _RAIter, _Compare) [with _RAIter = __gnu_cxx::__normal_iterator<std::tuple<double, int>*, std::vector<std::tuple<double, int> > >; _Compare = rankify(std::vector<double>&)::__lambda0]’
/usr/include/c++/4.8.2/bits/stl_algo.h:2332:68:   required from ‘void std::__introsort_loop(_RandomAccessIterator, _RandomAccessIterator, _Size, _Compare) [with _RandomAccessIterator = __gnu_cxx::__normal_iterator<std::tuple<double, int>*, std::vector<std::tuple<double, int> > >; _Size = long int; _Compare = rankify(std::vector<double>&)::__lambda0]’
/usr/include/c++/4.8.2/bits/stl_algo.h:5499:44:   required from ‘void std::sort(_RAIter, _RAIter, _Compare) [with _RAIter = __gnu_cxx::__normal_iterator<std::tuple<double, int>*, std::vector<std::tuple<double, int> > >; _Compare = rankify(std::vector<double>&)::__lambda0]’
blacklist.cpp:218:3:   required from here
/usr/include/c++/4.8.2/bits/stl_heap.h:313:40: error: no match for call to ‘(rankify(std::vector<double>&)::__lambda0) (std::tuple<double, int>&, std::tuple<double, int>&)’
        *(__first + (__secondChild - 1))))
                                        ^
blacklist.cpp:216:31: note: candidates are:
  std::sort(begin(T), end(T), [](auto const &t1, auto const &t2) {
                               ^
In file included from /usr/include/c++/4.8.2/bits/stl_algo.h:61:0,
                 from /usr/include/c++/4.8.2/algorithm:62,
                 from blacklist.cpp:18:
/usr/include/c++/4.8.2/bits/stl_heap.h:313:40: note: void (*)() <conversion>
        *(__first + (__secondChild - 1))))
                                        ^
/usr/include/c++/4.8.2/bits/stl_heap.h:313:40: note:   candidate expects 1 argument, 3 provided
blacklist.cpp:216:63: note: rankify(std::vector<double>&)::__lambda0
  std::sort(begin(T), end(T), [](auto const &t1, auto const &t2) {
                                                               ^
blacklist.cpp:216:63: note:   candidate expects 0 arguments, 2 provided
In file included from /usr/include/c++/4.8.2/algorithm:62:0,
                 from blacklist.cpp:18:
/usr/include/c++/4.8.2/bits/stl_algo.h: At global scope:
/usr/include/c++/4.8.2/bits/stl_algo.h:2110:5: error: ‘void std::__unguarded_linear_insert(_RandomAccessIterator, _Compare) [with _RandomAccessIterator = __gnu_cxx::__normal_iterator<std::tuple<double, int>*, std::vector<std::tuple<double, int> > >; _Compare = rankify(std::vector<double>&)::__lambda0]’, declared using local type ‘rankify(std::vector<double>&)::__lambda0’, is used but never defined [-fpermissive]
     __unguarded_linear_insert(_RandomAccessIterator __last,
     ^
In file included from /usr/include/c++/4.8.2/bits/stl_algo.h:61:0,
                 from /usr/include/c++/4.8.2/algorithm:62,
                 from blacklist.cpp:18:
/usr/include/c++/4.8.2/bits/stl_heap.h:331:5: error: ‘void std::__pop_heap(_RandomAccessIterator, _RandomAccessIterator, _RandomAccessIterator, _Compare) [with _RandomAccessIterator = __gnu_cxx::__normal_iterator<std::tuple<double, int>*, std::vector<std::tuple<double, int> > >; _Compare = rankify(std::vector<double>&)::__lambda0]’, declared using local type ‘rankify(std::vector<double>&)::__lambda0’, is used but never defined [-fpermissive]
     __pop_heap(_RandomAccessIterator __first, _RandomAccessIterator __last,
     ^
/usr/include/c++/4.8.2/bits/stl_heap.h:178:5: error: ‘void std::__push_heap(_RandomAccessIterator, _Distance, _Distance, _Tp, _Compare) [with _RandomAccessIterator = __gnu_cxx::__normal_iterator<std::tuple<double, int>*, std::vector<std::tuple<double, int> > >; _Distance = long int; _Tp = std::tuple<double, int>; _Compare = rankify(std::vector<double>&)::__lambda0]’, declared using local type ‘rankify(std::vector<double>&)::__lambda0’, is used but never defined [-fpermissive]
     __push_heap(_RandomAccessIterator __first, _Distance __holeIndex,
     ^
make: *** [all] Error 1

```

**Comment by aboyle** [comment]
Your C++ compiler does not support C++14 standard (-std=c++14), which is required. Please upgrade your compiler.
